modules/core/src/main/java/org/apache/synapse/rest/cors/SynapseCORSConfiguration.java

```diff
@@ -42,7 +50,30 @@ private SynapseCORSConfiguration() {
         //Retrieve allowed origin list
         String allowedOriginListStr =
                 SynapsePropertiesLoader.getPropertyValue(RESTConstants.CORS_CONFIGURATION_ACCESS_CTL_ALLOW_ORIGIN, "");
-        allowedOrigins = new HashSet<>(Arrays.asList(allowedOriginListStr.split(",")));
+        String[] originList = allowedOriginListStr.split(",");
+        for (String origin : originList) {
+            String trimmedOrigin = origin.trim();
+            allowedOrigins.add(trimmedOrigin);
+            try {
+                URL url = new URL(trimmedOrigin);
+                if (url.getHost().equals("localhost")) {
```

Comment:
shall we interchange the constant first in the if check to avoid NPEs

Reply:
NPE never happens since we create new instance just before the if block.